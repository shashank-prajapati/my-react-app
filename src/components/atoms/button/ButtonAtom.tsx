import React from "react";
import Button, { ButtonProps } from "@mui/material/Button";

interface Props extends ButtonProps {
  textColor?:string;
  backgroundColor?:string;
  hoverTextColor?:string;
  hoverBackGroundColor?:string;
  width?:string;
  hoverBottomBorder?:string;
  borderRadius?:string|number;
  ml?:string|number;
}

const ButtonAtom = (props: Props) => {
  const {textColor,backgroundColor,hoverTextColor,hoverBackGroundColor, children,width,hoverBottomBorder,borderRadius,ml,...rest } = props;
  return (
    <Button
      sx={{
        width:width,
        color:textColor,
        bgcolor:backgroundColor,
        borderRadius:borderRadius,
        ml:ml,
        '&:hover':{
          color:hoverTextColor,
          bgcolor:hoverBackGroundColor,
          borderBottom:hoverBottomBorder
        },
        textTransform: "none", fontWeight: "bold" }}
      {...rest}
    >{children}</Button>
  );
};

export default ButtonAtom;
