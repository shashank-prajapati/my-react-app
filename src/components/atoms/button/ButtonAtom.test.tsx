import * as React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import ButtonAtom from "./ButtonAtom";

describe("Testing Account Avatar", () => {
  it("should match the default snapshot", () => {
    const { container } = render(<ButtonAtom>Click Here</ButtonAtom>);
    expect(container).toMatchSnapshot();
  });

  it("should match the outlined button", () => {
    const { container } = render(
      <ButtonAtom variant="outlined">Click Here</ButtonAtom>
    );
    expect(container).toMatchSnapshot();
  });

  it("should match text snapshot", () => {
    const { container } = render(
      <ButtonAtom variant="text">Click Here</ButtonAtom>
    );
    expect(container).toMatchSnapshot();
  });

  it("should call on click method of button", () => {
    const handleClick = jest.fn();
    const { container } = render(
      <ButtonAtom onClick={handleClick}>Click Here</ButtonAtom>
    );
    fireEvent.click(screen.getByText("Click Here"));
    expect(handleClick).toHaveBeenCalled();
  });
});
