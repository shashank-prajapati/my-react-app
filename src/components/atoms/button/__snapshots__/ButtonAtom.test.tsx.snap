// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Account Avatar should match text snapshot 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-zm4jr-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Click Here
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Testing Account Avatar should match the default snapshot 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-zm4jr-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Click Here
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Testing Account Avatar should match the outlined button 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-ep45t5-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Click Here
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
